src/hooks/useSarcophagusNegotiation.ts

```diff
@@ -126,24 +136,57 @@ export function useSarcophagusNegotiation() {
         runSimulation.current = false;
 
         await createEncryptionKeypair();
-        const { encryptedShards, encryptedShardsTxId } = await uploadToArweave();
-
+        const { encryptedShards, encryptedShardsTxId } = await uploadAndSetEncryptedShards();
         console.log('initiating sarco negotiation');
         await initiateSarcophagusNegotiation(encryptedShards, encryptedShardsTxId);
       }
     })();
-  }, [publicKeysReady, dispatch, initiateSarcophagusNegotiation, runSimulation, createEncryptionKeypair, uploadToArweave]);
+  }, [
+    publicKeysReady,
+    dispatch,
+    initiateSarcophagusNegotiation,
+    runSimulation,
+    createEncryptionKeypair,
+    uploadAndSetEncryptedShards,
+  ]);
 
+  // TODO: move this to its own hook and refactor when create sarcophagus flow is finalized
   // Simulate call to create sarco when signatures ready
   // Likely scenario is to use signaturesReady to visually prompt user to
   // click that final submit button to make the contract call.
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [isApproving, setIsApproving] = useState(false);
+  const { approve } = useApprove();
+  const { allowance } = useAllowance();
+
   useEffect(() => {
     if (signaturesReady && runCreateSimulation.current) {
       runCreateSimulation.current = false;
       console.log('signatures ready');
+      uploadAndSetDoubleEncryptedFile();
+    }
+
+    if (!!payloadTxId && !allowance?.gt(0) && !isApproving) {
+      setIsApproving(true);
+      approve();
+    }
+
+    if (!!payloadTxId && !isSubmitting && allowance?.gt(0)) {
+      setIsSubmitting(true);
+      console.log('file upload to arweave complete, submitting create sarcophagus');
```

Comment:
Much of this `useEffect` is a placeholder, and the allowance workflow will need to be setup in the UX.

I'm also leaving some extra console logging in the codebase for debugging purposes -- this can eventually get removed.